libexec/drivers/coffeetags

```diff
@@ -27,9 +27,13 @@
 #
 #
 
-if ! type coffeetags > /dev/null 2>&1; then
-    exit 77
-fi
+{
+    if  [[ (! $(type coffeetags)) || \
+        ($(type rbenv) && ! $(rbenv which coffeetags)) ]]
```

Comment:
b4n, you've made a number of very good points, so let me address them in order:

> `[[ ]]` is a bashism, and this script uses `/bin/sh`

You are absolutely right. This is an oversight on my part, so I'll update the PR to be POSIX compliant.

> Someone could be running outside of rbenv yet have `rbenv` in their PATH

This is also true. The way rbenv works is that after it is initialised `rbenv` is available as a function. Therefore, testing for the presence of such a function (eg: `[ "$(type -t rbenv)" = "function" ]`) would give us a reliable way to identify active rbenv instances. I'll update the PR to reflect this change.

> What if both `type rbenv` and `rbenv which coffeetags` succeed but `type coffeetags` does not?

When initialises, rbenv creates a surrogate for every executable installed by any version of ruby it manages. These surrogates are updates every time rbenv changes current version of ruby in such a way that they always point to either a real executable or a generic notice stating that this executable is currently unavailable. Thus, I can't think of a single scenario where `rbenv which coffeetags` could succeed but `type coffeetags` would fail.

However, safeguarding ourselves from this unlikely scenario does us no harm, and saves us from a potential headache later, if rbenv ever changes its current behaviour. So, once again, I'll update the PR. :)